charleshu-8/WebDevProject
# Create control panel

```git switch create-control-panel```
run ```pnpm dev```
then click on go to chat

What was created: 
- skeleton for chatpage containing the side panel, fixed spacing/layout for all elements on chat page, and added container (Box) for all future discussion form elements
- Nested SidePanel() function into the SideBar component so the two can share data in order to switch between the different panels
- Implemented the Panel Siwtiching functionality so when you click on the sidebar buttons you see the different panels

What I fixed:

- so since the NavBar is set to fixed positioning --> this means it was taken out of the flow of the html DOM, so other components won't respond to its presence
- so to fix this: i made sure to change all elements below nav bar positioning to relative to move all elements down in a container Box where top is set at 80px down (height of navbar) and height is adjusted using calc() function to account for parent height - navbar height
- container is set up for adding discussion content below it, so any future elements added to page shouldn't conflict with navbar positioning if their height is set to full
- fixed overflow viewport bug so now the chat page should not be scrollable at all

How to test: 
- Inspect chat page: page.tsx elements 
-- click F12 on windowns and fn + F12 on mac to pull up developer console and click the leftmost button on upper toolbar in console to inspect elements. 
- Resize window browser to see how elements respond
- click on sidebar buttons and see how the panel changes

Code review of check positioning, look at files in chat folder: 
-sidebar.tsx, sidepanel.tsx, page.tsx
- also had to fix chat-input-field since updated code unifiaction/styligns weren't added when merged
